chasm/lib/activity/statemachine.go

```diff
@@ -33,12 +41,48 @@ var TransitionScheduled = chasm.NewTransition(
 			return err
 		}
 
+		currentTime := ctx.Now(a)
+		retryInterval := time.Duration(0)
+		isRetry := attempt.GetCount() >= 1
+		attempt.Count += 1
+
+		// If this is a retry, calculate the delay before scheduling tasks and update attempt fields
+		if isRetry {
+			retryInterval = backoff.CalculateExponentialRetryInterval(a.GetRetryPolicy(), attempt.GetCount())
+			attempt.CurrentRetryInterval = durationpb.New(retryInterval)
```

Comment:
You might want to put a TODO here to set this to the next interval from the activity failure on a failed attempt.